optaplanner-core/src/test/java/org/optaplanner/core/impl/heuristic/selector/entity/pillar/DefaultPillarSelectorTest.java

```diff
@@ -312,11 +312,10 @@ public void randomWithSubs_Size2() {
         // nextInt pattern: pillarIndex, subPillarSize, element 0, element 1, element 2, ...
         // Expected pillar cache: [a], [b, d], [c, e, f]
         when(workingRandom.nextInt(anyInt())).thenReturn(
-                0, 0, 0, // [a] If the pillar is less than the minimum, the minimum is reduced
-                2, 0, 0, 0, // [c, e, f]
-                1, 0, 0, 0, // [b, d]
-                1, 0, 1, 0); // [b, d]
-        assertCodesOfNeverEndingPillarSelector(pillarSelector, "[a]", "[c, e]", "[b, d]", "[d, b]");
+                0, // [a]
+                2, 0, 0, 1, // [c, e, f]
+                1, 0); // [b, d]
```

Comment:
The last option (`[d, b]` instead of `[b, d]`) was removed, as one of the shortcuts makes it impossible to generate this permutation. It has been agreed beforehand that removing permutations is not an issue.